scheduler/README.md

```diff
@@ -0,0 +1,192 @@
+┌────────────────────────────┐  "true" if deployment is complete
+│computeDeploymentComplete(m)│  for the task group.
+└────────────────────────────┘  computeDeploymentComplete itself
+               │                returns a boolean.
+               │
+               │              ┌────────────────────────────────────┐
+               │              │    computeGroup(groupName, all     │
+               ├─────────────▶│            allocations)            │
+               │              └────────────────────────────────────┘
+               │
+               │                contains the main, and most
+               │                complex part of the reconciler.
+               │                it calls many helper methods:
+               │                - filterOldTerminalAllocs: allocs
+               │                that are terminal or from older
+               │                job ver are put into "ignore"
+               │                bucket
+               │                - cancelUnneededCanaries
+               │                - filterByTainted: results in 6
+               │                buckets mentioned in the
+               │                paragraphs above: untainted,
+               │                migrate, lost, disconnecting,
+               │                reconnecting, ignore and
+               │                expiring.
+               └───────┐        - filterByRescheduleable: updates
+                       │        the untainted bucket and creates
+                       │        2 new ones: rescheduleNow and
+                       │        rescheduleLater
+                       │        - reconcileReconnecting: returns
+                       │        which allocs should be marked for
+                       │        reconnecting and which should be
+                       │        stopped
+                       │        - computeStop
+                       │        - computeCanaries
+                       │        - computePlacements: allocs are
+                       │        placed if deployment is not
+                       │        paused or failed, they are not
+                       │        canaries (unless promoted),
+                       │        previous alloc was lost
+                       │        - computeReplacements
+                       │        - createDeployment
+                       │
+                       ▼
+┌────────────────────────────────────────────┐
+│computeDeploymentUpdates(deploymentComplete)│
+└────────────────────────────────────────────┘
+                       │
+                ┌──────┘        for complete deployments, it
+                │               handles multi-region case and
+                ▼               sets the deploymentUpdates
+        ┌───────────────┐
+        │return a.result│
+        └───────────────┘
+```
+
+## Feasibility checking
+
+Nomad uses a set of iterators to iterate over all nodes and check how feasible
```

Comment:
> all nodes

We're skipping over all the logic about the initial filter by eligibility, DC, and node pool that happens before the stack. And we're skipping over the logic of the limit iterator that means we don't process _all_ the nodes. That's all probably ok for this pass but let's keep that in mind for future iterations.